packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -191,14 +191,17 @@ Future<int> runInteractively(List<String> command, {
     allowReentrantFlutter: allowReentrantFlutter,
     environment: environment,
   );
-  process.stdin.addStream(stdin);
+  // The real stdin will never finish streaming. Pipe until the child process
+  // finishes.
+  process.stdin.addStream(stdin); // ignore: unawaited_futures
   // Wait for stdout and stderr to be fully processed, because process.exitCode
   // may complete first.
   await Future.wait<dynamic>(<Future<dynamic>>[
     stdout.addStream(process.stdout),
     stderr.addStream(process.stderr),
   ]);
-  return await process.exitCode;
+  final int exitCode = await process.exitCode;
```

Comment:
Why not just `return await process.exitCode;` as it was before?

Reply:
Oops, I was trying something else and forgot to revert everything